src/ConstructiveSolidGeometry/plotting/SurfacePrimitives/RegularPolygon.jl

```diff
@@ -0,0 +1,29 @@
+function get_plot_points(rp::RegularPolygon{N,T}; kwargs...) where {N, T <: AbstractFloat}
+
+    plot_points = Vector{CartesianPoint{T}}[]
+
+    rMin::T = _left_radial_interval(rp.r)
+    rMax::T = _right_radial_interval(rp.r)
+
+    for r in (rMin == 0 ? [rMax] : [rMin, rMax])
+        push!(plot_points, Vector{CartesianPoint{T}}([CartesianPoint{T}(r * cos(φ), r * sin(φ), rp.z) for φ in range(0,2π,length=N+1)]))
+    end
+
+    plot_points
+end
+
+function mesh(rp::RegularPolygon{N,T}; n = 30) where {N, T <: AbstractFloat}
+
+    rMin::T, rMax::T = get_r_limits(rp)
+    φrange = range(0, 2π, length = N+1)
+    sφrange = sin.(φrange)
+    cφrange = cos.(φrange)
```

Comment:
You can try `scφrange = sincos.(φrange)` here and later use `for (sφ,cφ) in scφrange`
(but I do not know if that is faster..)